pandas/core/reshape/melt.py

```diff
@@ -32,7 +32,11 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
             raise ValueError('id_vars must be a list of tuples when columns'
                              ' are a MultiIndex')
         else:
+            # Check that `id_vars` are in frame
             id_vars = list(id_vars)
+            missing = [v for v in id_vars if v not in frame.columns]
+            if missing:
+                raise ValueError(f'Columns {missing} are not in dataframe')
```

Comment:
We support older pythons, so we can't use f-strings yet unfortunately.

Reply:
ok, I can do it with `%s` string substitutions or is best practice just to give a general error?